src/modules/rlm_python/rlm_python.c

```diff
@@ -205,6 +206,56 @@ static void mod_error(void)
 	Py_XDECREF(pTraceback);
 }
 
+static int dlopen_linked_libpython_cb(struct dl_phdr_info *info,
+					UNUSED size_t size, void *data)
+{
+	const char *pattern = "/libpython"
+				STRINGIFY(PY_MAJOR_VERSION) "."
+				STRINGIFY(PY_MINOR_VERSION) ".so";
+	char **ppath = (char **)data;
+
+	if (strstr(info->dlpi_name, pattern) != NULL) {
+		if (*ppath != NULL) {
+			free(*ppath);
+			*ppath = NULL;
+			return EEXIST;
+		} else {
+			*ppath = strdup(info->dlpi_name);
```

Comment:
It's better to use talloc functions, because it allows us to track all the allocations done by us when debugging memory leaks.